### PR TITLE
Task-4 | Display recipes 

### PR DESCRIPTION
**Pull request by KiraSl**
This PR applies the changes from the following [Task](https://github.com/KiraSl/react-recipes/issues/4)

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -27,42 +29,14 @@ class App extends React.Component {
...
-            <div className="col-sm-6 col-lg-4 mb-4">
-              <div className="card">
-                <img src="..." className="card-img-top" alt="..." />
-                <div className="card-body">
-                  <h5 className="card-title">Card title</h5>
-                  <p className="card-text">Some quick example text to build on the card title and make up the bulk of the card’s content.</p>
-                </div>
-              </div>
-            </div>
+            {recipes.map((recipe, i) =>
```
Comment (samdemaeyer) [inline src/App.js:32]: You receive the recipe here, there is no need to do another lookup, you can use it straight away from here.
eg. `imageUrl={recipe.image}`

File: src/App.js [hunk 2]
```diff
@@ -27,42 +29,14 @@ class App extends React.Component {
...
+              <RecipeCard
+                imageUrl={recipes[i].image}
+                title={recipes[i].title}
+                description={recipes[i].description}
+                key={i}
+              />
```
Comment (samdemaeyer) [inline src/App.js:38]: This is ok, as you probably want to see clearly what is used inside the card. Just want to point out that you don't have to pass every attr separately. You could have passed the card object as a whole.  
```js
<RecipeCard 
  reciepe={recipe} 
  key={i} 
/>
```
In this case, if your `RecipeCard` component grows, you don't need to add more and more attrs. If your application grows, you can easily end up with a big amount of attrs. 
For example 👇 
```js
<RecipeCard
  id={recipe.id}
  imageUrl={recipe.image}
  smalImageUrl={recipe.smalImageUrl}
  title={recipe.title}
  subTitle={recipe.subTitle}
  description={recipe.description}
  foo={recipe.foo}
  bar={recipe.bar}
  type={type}
  size={size}
  icon={icon}
  key={i}
/>
```
Note that `type`, `size` and `icon` are attributes for the card. All the rest are nested inside the `recipe` object.
You could easily do 👇 
```js
<RecipeCard
  recipe={recipe}
  type={type}
  size={size}
  icon={icon}
  key={i}
/>
```

**Comment by samdemaeyer** [comment]
@KiraSl, I forgot to add 1 step to the task, I just added it. Can you try with bootstrap classes to make sure the height of each card is the same?

**Before:**
Screenshot 2020-06-07 at 11 34 40

**After:**
Screenshot 2020-06-07 at 11 37 58

File: src/App.js (hunk 1 above)
Comment (KiraSl) [reply src/App.js:32]: can't believe I actually did it this way 🤦

File: src/RecipeCard.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+
+const propTypes = {
+  recipe: PropTypes.object.isRequired,
```
Comment (samdemaeyer) [inline src/RecipeCard.js:5]: I would still validate the properties here 👇 
```suggestion
  recipe: PropTypes.shape({
    image: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
  }),
```

**Pull request merged**